test_apis/test_merchant.py

```diff
@@ -0,0 +1,206 @@
+                    'While comparing JSON an exception occured: {exception}\n'.format(
+                        exception=e))
+            assert True
+            file_pointer.write(
+                'Test ends for {name} at {datetime}'.format(datetime=datetime.now(), name=self.name_merchant_outlets_with_id))
+            file_pointer.close()
+
+    def test_with_all_correct_parameters_merchant_name(self):
+
+        file_pointer = open('logs/{name}_{date}.txt'.format ( date=self.date, name=self.name_merchant_name), 'w')
+        file_pointer.write('Going to test {name} with correct data at {datetime}\n'.format(
+            datetime=datetime.now(), name=self.name_merchant_name))
+
+        response_php = requests.get(PHP_LINK + self.url_merchant_name, auth=self.http_auth,
+                                      headers=self.header, )
+        response_python = requests.get(PYTHON_LINK + self.url_merchant_name, auth=self.http_auth,
+                                         headers=self.header, )
+
+        file_pointer.write(
+            '{response_php} is php response status code and {response_python} is python response status code\n'.format(
+                response_php=response_php.status_code, response_python=response_python.status_code))
+        try:
+            compare_data_dictionaries(response_php.json(), response_python.json(), file_pointer)
+        except Exception as e:
+            file_pointer.write(
+                'While comparing JSON an exception occured: {exception}\n'.format(
+                    exception=e))
+        assert True
+        file_pointer.write(
+            'Test ends for {name} at {datetime}'.format(datetime=datetime.now(), name=self.name_merchant_name))
+        file_pointer.close()
+
+    def test_with_all_correct_parameters_merchant_names(self):
+
+        file_pointer = open('logs/{name}_{date}.txt'.format(date=self.date, name=self.name_merchant_names ), 'w')
+        file_pointer.write('Going to test {name} with correct data at {datetime}\n'.format(
+            datetime=datetime.now(), name=self.name_merchant_names))
+
+        response_php = requests.get(PHP_LINK + self.url_merchant_names, auth=self.http_auth,
+                                      headers=self.header, )
+        response_python = requests.get(PYTHON_LINK + self.url_merchant_names, auth=self.http_auth,
+                                         headers=self.header,)
+
+        file_pointer.write(
+            '{response_php} is php response status code and {response_python} is python response status code\n'.format(
+                response_php=response_php.status_code, response_python=response_python.status_code))
+        try:
+            compare_data_dictionaries(response_php.json(), response_python.json(), file_pointer)
+        except Exception as e:
+            file_pointer.write(
+                'While comparing JSON an exception occured: {exception}\n'.format(
+                    exception=e))
+        assert True
```

Comment:
Response data can be converted to the json format then main data can be written separately.